Mathlib/CategoryTheory/Monoidal/OfChosenFiniteProducts/Symmetric.lean

```diff
@@ -1,95 +1,3 @@
-
-open CategoryTheory.Limits
-
-variable (𝒯 : LimitCone (Functor.empty.{0} C))
-variable (ℬ : ∀ X Y : C, LimitCone (pair X Y))
-
-open MonoidalOfChosenFiniteProducts
-
-namespace MonoidalOfChosenFiniteProducts
-
-open MonoidalCategory
-
-theorem braiding_naturality {X X' Y Y' : C} (f : X ⟶ Y) (g : X' ⟶ Y') :
-    tensorHom ℬ f g ≫ (Limits.BinaryFan.braiding (ℬ Y Y').isLimit (ℬ Y' Y).isLimit).hom =
-      (Limits.BinaryFan.braiding (ℬ X X').isLimit (ℬ X' X).isLimit).hom ≫ tensorHom ℬ g f := by
-  dsimp [tensorHom, Limits.BinaryFan.braiding]
-  apply (ℬ _ _).isLimit.hom_ext
-  rintro ⟨⟨⟩⟩ <;> · dsimp [Limits.IsLimit.conePointUniqueUpToIso]; simp
-
-theorem hexagon_forward (X Y Z : C) :
-    (BinaryFan.associatorOfLimitCone ℬ X Y Z).hom ≫
-        (Limits.BinaryFan.braiding (ℬ X (tensorObj ℬ Y Z)).isLimit
-              (ℬ (tensorObj ℬ Y Z) X).isLimit).hom ≫
-          (BinaryFan.associatorOfLimitCone ℬ Y Z X).hom =
-      tensorHom ℬ (Limits.BinaryFan.braiding (ℬ X Y).isLimit (ℬ Y X).isLimit).hom (𝟙 Z) ≫
-        (BinaryFan.associatorOfLimitCone ℬ Y X Z).hom ≫
-          tensorHom ℬ (𝟙 Y) (Limits.BinaryFan.braiding (ℬ X Z).isLimit (ℬ Z X).isLimit).hom := by
-  dsimp [tensorHom, Limits.BinaryFan.braiding]
-  apply (ℬ _ _).isLimit.hom_ext; rintro ⟨⟨⟩⟩
-  · dsimp [Limits.IsLimit.conePointUniqueUpToIso]; simp
-  · apply (ℬ _ _).isLimit.hom_ext
-    rintro ⟨⟨⟩⟩ <;> · dsimp [Limits.IsLimit.conePointUniqueUpToIso]; simp
-
-theorem hexagon_reverse (X Y Z : C) :
-    (BinaryFan.associatorOfLimitCone ℬ X Y Z).inv ≫
-        (Limits.BinaryFan.braiding (ℬ (tensorObj ℬ X Y) Z).isLimit
-              (ℬ Z (tensorObj ℬ X Y)).isLimit).hom ≫
-          (BinaryFan.associatorOfLimitCone ℬ Z X Y).inv =
-      tensorHom ℬ (𝟙 X) (Limits.BinaryFan.braiding (ℬ Y Z).isLimit (ℬ Z Y).isLimit).hom ≫
-        (BinaryFan.associatorOfLimitCone ℬ X Z Y).inv ≫
-          tensorHom ℬ (Limits.BinaryFan.braiding (ℬ X Z).isLimit (ℬ Z X).isLimit).hom (𝟙 Y) := by
-  dsimp [tensorHom, Limits.BinaryFan.braiding]
-  apply (ℬ _ _).isLimit.hom_ext; rintro ⟨⟨⟩⟩
-  · apply (ℬ _ _).isLimit.hom_ext
-    rintro ⟨⟨⟩⟩ <;>
-      · dsimp [BinaryFan.associatorOfLimitCone, BinaryFan.associator,
-          Limits.IsLimit.conePointUniqueUpToIso]
-        simp
-  · dsimp [BinaryFan.associatorOfLimitCone, BinaryFan.associator,
-      Limits.IsLimit.conePointUniqueUpToIso]
-    simp
-
-theorem symmetry (X Y : C) :
-    (Limits.BinaryFan.braiding (ℬ X Y).isLimit (ℬ Y X).isLimit).hom ≫
-        (Limits.BinaryFan.braiding (ℬ Y X).isLimit (ℬ X Y).isLimit).hom =
-      𝟙 (tensorObj ℬ X Y) := by
-  dsimp [tensorHom, Limits.BinaryFan.braiding]
-  apply (ℬ _ _).isLimit.hom_ext
-  rintro ⟨⟨⟩⟩ <;> · dsimp [Limits.IsLimit.conePointUniqueUpToIso]; simp
-
-end MonoidalOfChosenFiniteProducts
-
-open MonoidalOfChosenFiniteProducts
-
-/-- The monoidal structure coming from finite products is symmetric.
--/
-def symmetricOfChosenFiniteProducts :
-    SymmetricCategory (MonoidalOfChosenFiniteProductsSynonym 𝒯 ℬ) where
```

Comment:
I think this result got lost?

Reply:
No, it's already available by typeclass inference. See [`CategoryTheory.ChosenFiniteProducts.instSymmetricCategoryOfBraidedCategory`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/CategoryTheory/ChosenFiniteProducts.html#CategoryTheory.ChosenFiniteProducts.instSymmetricCategoryOfBraidedCategory). `MonoidalOfChosenFiniteProductsSynonym` was described as an auxiliary type, so I simply deleted it without deprecation

Reply:
Ah sure, but once you assume `BraidedCategory` (eg using [`CategoryTheory.BraidedCategory.ofChosenFiniteProducts`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/CategoryTheory/ChosenFiniteProducts.html#CategoryTheory.BraidedCategory.ofChosenFiniteProducts)) then you automatically get `SymmetricCategory`. So all functionality is still here

Reply:
Right, but that's exactly where you get diamonds from. The point of the existing setup is to state `symmetricOfChosenFiniteProducts` without getting diamonds or defeq issues at all, and it's not clear to me that the new way preserve this?

Reply:
Nonono

`BraidedCategory` just isn't part of `ChosenFiniteProducts`. It's a mixin over `MonoidalCategory`. All that's happening is that, in the presence of `ChosenFiniteProducts C`, `BraidedCategory C` gets upgraded to `SymmetricCategory C` (this doesn't create data)

Reply:
discussed offline, I'm content